core/unix/signal.c

```diff
@@ -6414,6 +6411,20 @@ os_forge_exception(app_pc target_pc, dr_exception_type_t type)
 #if defined(LINUX) && defined(X86)
     /* We use a TLS buffer to avoid too much stack space here. */
     sc->fpstate = (kernel_fpstate_t *)get_xstate_buffer(dcontext);
+    sc->fpstate->sw_reserved.extended_size = signal_frame_extra_size(false);
+    if (YMM_ENABLED()) {
+        sc->fpstate->sw_reserved.magic1 = FP_XSTATE_MAGIC1;
```

Comment:
Not sure. Apparently this causes a failure of test linux.bad-signal-stack, where the app's altstack is read-only. The test wants a SIGSEGV delivered. Does this somehow cause us to write into this memory before and we crash? But didn't we do that before already?

Reply:
The bug fix is to memset the thing to 0 first (this is a heap-allocated synthetic xstate).  It seems to be having non-zero values in the reserved1 field of kernel_xsave_hdr_t that causes the crash: it seems to cause the kernel to change the %gs base.  Kind of strange, but good to not have uninitialized values in there.

Reply:
(Will put a comment by the memset.)